docs/decisions/0002-pie-structure.rst

```diff
@@ -0,0 +1,87 @@
+1. Structure of the Program Intent Engagement Service
+=====================================================
+
+Status
+------
+
+Accepted
+
+What is a Program Intent?
+-------------------------
+
+A program intent records a **learner's** **intent** to complete a **program**. That intent can be positive or negative, certain or uncertain. Each intent also records the **reason** we are storing the intent, such as package purchase or completion of courses within the program.
+
+Intents are never modified. If new information gives us a new belief about the learner's intentions, we write a new intent for that learner and program, leaving the old one in place. Intents have two timestamps to assist with ordering, the standard **created** timestamp and an **effective** timestamp which is the time this intent became true to the system. If a weekly batch job on Saturday were to notice that the user had signed up for a program on Tuesday, created would be Saturday, and effective would be Tuesday.
+
+Model
++++++
+
++--------------------+----------------------------------------+
+|Data                |Note                                    |
++====================+========================================+
+|Program ID          |program UUID is the generic identifier  |
```

Comment:
This has a relationship to the program UUID in discovery. Is that worth calling out here or too much an implementation detail?

In any case, what do you mean by "generic" here?